dir/path.go

```diff
@@ -79,21 +79,35 @@ const (
 
 var userConfigDir = os.UserConfigDir // for unit test
 
-func init() {
-	loadUserPath()
-}
-
 // loadUserPath function defines UserConfigDir and UserLibexecDir.
 func loadUserPath() {
-	// set user config
-	userDir, err := userConfigDir()
-	if err != nil {
-		panic(err)
+	UserConfigDir = ""
+	UserLibexecDir = ""
+
+	_ = UserLibexecDirPath()
+}
+
+// UserConfigDirPath returns the user level {NOTATION_CONFIG} path.
+func UserConfigDirPath() string {
+	if UserConfigDir == "" {
+		userDir, err := userConfigDir()
+		if err != nil {
+			// fallback to current directory
+			userDir = "."
+		}
+		// set user config
+		UserConfigDir = filepath.Join(userDir, notation)
 	}
-	UserConfigDir = filepath.Join(userDir, notation)
+	return UserConfigDir
+}
 
-	// set user libexec
-	UserLibexecDir = UserConfigDir
+// UserLibexecDirPath returns the user level {NOTATION_LIBEXEC} path.
+func UserLibexecDirPath() string {
```

Comment:
```suggestion
func userLibexecDirPath() string {
```
Can we set it as an internal function? If use want to get the config dir, they can call `PluginFS().SysPath()`

Reply:
Resolved in [5281e07](https://github.com/notaryproject/notation-go/pull/449/commits/5281e07b3601e70fe467b000158ea7862e44b713).